File: November LeetCoding Challenge 2021/1413. Minimum Value to Get Positive Step by Step Sum.cpp
class Solution {
public:
    int minStartValue(vector<int>& nums) {
        int l = 1, r = 1e4+123, ans = 1e4+123;
        
        while ( l <= r ) {
            int mid = ( l + r ) >> 1;
            int tmp = mid;
            
            bool done = 1;
            for ( auto u : nums ) {
                mid += u;
                if ( mid < 1 ) {
                    done = 0;
                    break;
                }
            }
            
            if ( done ) {
                ans = min ( ans, tmp );
                r = tmp-1;
            }
            else l = tmp+1;
        }
        
        return ans;
    }
};

File: Solution with python/November LeetCoding Challenge 2021/1413. Minimum Value to Get Positive Step by Step Sum.py
class Solution(object):
    def minStartValue(self, nums):
        
        l = 1
        r = 10000+123
        ans = 10000+123
        while ( l <= r ) :
            mid = ( l + r ) >> 1
            tmp = mid
            done = 1
            
            for u in nums :
                tmp += u
                if tmp < 1 :
                    done = 0
                    break
                    
                    
            if done == 1 :
                ans = min ( ans, mid )
                r = mid-1
                
            else :
                l = mid+1
                
        return ans;
        